pkg/controlplane/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go

```diff
@@ -256,32 +271,25 @@ func getConfigMapDataFor(authenticationInfo ClusterAuthenticationInfo) (map[stri
 		return data, nil
 	}
 
-	if caBytes := authenticationInfo.RequestHeaderCA.CurrentCABundleContent(); len(caBytes) > 0 {
-		var err error
-
-		// encoding errors aren't going to get better, so just fail on them.
-		data["requestheader-username-headers"], err = jsonSerializeStringSlice(authenticationInfo.RequestHeaderUsernameHeaders.Value())
-		if err != nil {
-			return nil, err
-		}
-		if utilfeature.DefaultFeatureGate.Enabled(features.RemoteRequestHeaderUID) && len(authenticationInfo.RequestHeaderUIDHeaders.Value()) > 0 {
-			data["requestheader-uid-headers"], err = jsonSerializeStringSlice(authenticationInfo.RequestHeaderUIDHeaders.Value())
-			if err != nil {
-				return nil, err
-			}
-		}
+	caBytes := authenticationInfo.RequestHeaderCA.CurrentCABundleContent()
+	if len(caBytes) == 0 {
+		return data, nil
+	}
+	data["requestheader-client-ca-file"] = string(caBytes)
 
-		data["requestheader-group-headers"], err = jsonSerializeStringSlice(authenticationInfo.RequestHeaderGroupHeaders.Value())
-		if err != nil {
-			return nil, err
-		}
-		data["requestheader-extra-headers-prefix"], err = jsonSerializeStringSlice(authenticationInfo.RequestHeaderExtraHeaderPrefixes.Value())
-		if err != nil {
-			return nil, err
+	var err error
+	for key, configData := range map[string][]string{
+		"requestheader-username-headers":     authenticationInfo.RequestHeaderUsernameHeaders.Value(),
+		"requestheader-uid-headers":          authenticationInfo.RequestHeaderUIDHeaders.Value(),
```

Comment:
why is it safe to unconditionally set this key again? won't that duel with 1.32 API servers?

Reply:
I'll remove the UIDs beta move from this PR, let's focus on handling the existing keys behavior first.